packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts

```diff
@@ -221,6 +221,20 @@ export interface QueueProcessingServiceBaseProps {
    * @default - false
    */
   readonly enableExecuteCommand?: boolean;
+
+  /**
+   * Whether CPU based scaling of container should be enabled should be enabled.
+   *
+   * @default - false
+   */
+  readonly enableCpuBasedScaling?: boolean;
+
+  /**
+   * CPU based scaling target utilization value
+   *
+   * @default 50
+   */
+  readonly cpuBasedScalingTargetUtilization?: number;
```

Comment:
Alternatively instead of the enableCpuBasedScaling flag, we can combine both these new props by checking the number passed. If 1-100 then it is enabled at that value. If -1 then cpu utilization is disabled. And then instead of checking on true/false flag then we can check on the numbers value.

If we keep the current implementation then we will need to check for the situation a customer passes 
`enableCpuBasedScaling: false` and `cpuBasedScalingTargetUtilization: <a_valid_int>` and then throw an error as this is not a valid prop configuration.